### ISSUE TITLE
sharepoint will be omitted from the output because of failed authentication

### ISSUE THREAD

**Issue by lexpr0** [cisagov/ScubaGear#1407]
After creating a new tenant to test Scuba, I run the command:

Invoke-SCuBA -ProductNames aad, defender, exo, powerplatform, sharepoint, teams

After loging in as a user with the permissions set as desribed (but also as the global admin) all tests run fine, except for Sharepoint. I receive the error: "sharepoint will be omitted from the output because of failed authentication"

## To reproduce ##

1. Create a new tenant, set the permissions as described.
2. Run Invoke-SCuBA 
3. use the created user (with the permissions as described) of tenant admin.

## Expected behavior ##
I expect the Sharepoint test to run like the other tests.

## Any helpful log output or screenshots ##
Connect-Tenant : Error establishing a connection with sharepoint. Could not connect to SharePoint Online.
At C:\Program Files\WindowsPowerShell\Modules\ScubaGear\1.3.0\Modules\Orchestrator.psm1:1187 char:26
+         $AnyFailedAuth = Connect-Tenant @ConnectTenantParams
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Connect-Tenant

($PSVersionTable).PsVersion
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      22621  4391

 (Get-Module -ListAvailable -Name Microsoft.Online.SharePoint.PowerShell).Version
Major  Minor  Build  Revision
-----  -----  -----  --------
16     0      24810  12000

The users has the Sharepoint administrator role, the other account is global admin, but both give the same result.

**Comment by lexpr0** [comment]
The command seems to work, I am not receiving any errors.

**Comment by lexpr0** [comment]
> > The command seems to work, I am not receiving any errors.
> > [@lexpr0](https://github.com/lexpr0)
> 
> Thanks! We'll include the fix in a future release.
> 
> In the meantime, to bypass the error, running ScubaGear with [service principal auth](https://github.com/cisagov/ScubaGear/blob/main/docs/prerequisites/noninteractive.md) will bypass the error.

I am trying to integratie your fix, adding the paramters:
                                   'Url'= "https://$($InitialDomainPrefix)-admin.sharepoint.com";
				   'ModernAuth'=  $true;
				   'AuthenticationUrl'=  "https://login.microsoftonline.com/organizations";
                               }

However I get some simalar as albber-widjaja:
Connect-Tenant : Error establishing a connection with sharepoint. Could not connect to SharePoint Online.
At C:\Program Files\WindowsPowerShell\Modules\ScubaGear\1.3.0\Modules\Orchestrator.psm1:1187 char:26
+         $AnyFailedAuth = Connect-Tenant @ConnectTenantParams
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Connect-Tenant

I have tried Initialize-SCuBA but without succes.

**Comment by lexpr0** [comment]
Correct, I have added the parameter as follows:

`        {($_ -eq "commercial") -or ($_ -eq "gcc")} {

                               $SPOParams += @{

                                   'Url'= "https://$($InitialDomainPrefix)-admin.sharepoint.com";

				   'ModernAuth'=  $true;

				   'AuthenticationUrl'=  "https://login.microsoftonline.com/organizations";`

**Comment by lexpr0** [comment]
> [@lexpr0](https://github.com/lexpr0) Did [@buidav](https://github.com/buidav) 's latest suggestion to use the latest release or upgrading the SharePoint modules resolve your issue?

No, the issue was not resolved. However this issue was only relevant with a newly created tenant.

**Comment by paulof-adaquest** [comment]
Hello, it's an old tread.

For me worked when I executed the following command "Disconnect-SCuBATenant"

I had this issue due connection with other Tenant at the first connection. After execute the command, I call invoke-scuba again and it worked.

**Comment by FollyBeachGurl** [comment]
Hello @lexpr0. I am reaching out to see if you have made any current updates on this issue.  Trying to determine the current status of this issue.

**Comment by FollyBeachGurl** [comment]
Hello @lexpr0. I am reaching out again to see if you have made any current updates on this issue. Is this still true? I am trying to determine the current status of this issue.

**Comment by FollyBeachGurl** [comment]
Reached out twice with no response. 
Closing this issue.
